src/storage/types.rs

```diff
@@ -105,7 +105,7 @@ pub struct Fingerprint {
     pub card_fingerprint: Secret<String>,
 }
 
-#[derive(serde::Serialize, Debug)]
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
```

Comment:
This i think will conflict with @ShankarSinghC 's PR, but everything else looks good!